File: spring-data/mongodb-api/src/main/java/com/github/mehdishahdoost/mongodbapi/repository/StudentRepository.java
package com.github.mehdishahdoost.mongodbapi.repository;

import com.github.mehdishahdoost.mongodbapi.model.Student;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface StudentRepository  extends MongoRepository<Student, String> {
    Optional<Student> findByEmail(String email);
}

File: mongodb-api-kotlin/src/main/kotlin/com/github/mehdishahdoost/repository/StudentRepository.kt
package com.github.mehdishahdoost.repository

import com.github.mehdishahdoost.model.Student
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface StudentRepository : MongoRepository<Student, String> {
    fun findByEmail(email: String): Student
}